File: app/src/main/java/org/antrack/app/libs/SessionIdGenerator.java
package org.antrack.app.libs;

// http://stackoverflow.com/a/41156/5984995

import java.math.BigInteger;
import java.security.SecureRandom;

public final class SessionIdGenerator {
    private SecureRandom random = new SecureRandom();

    public String nextSessionId() {
        return new BigInteger(130, random).toString(32);
    }
}

File: app/src/main/java/org/antrack/app/libs/SessionIdGenerator.kt
package org.antrack.app.libs

// http://stackoverflow.com/a/41156/5984995

import java.math.BigInteger
import java.security.SecureRandom

class SessionIdGenerator {
    private val random = SecureRandom()

    fun nextSessionId(): String {
        return BigInteger(130, random).toString(32)
    }
}
